staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -337,7 +325,27 @@ func NewCacherFromConfig(config Config) *Cacher {
 		bookmarkWatchers:     newTimeBucketWatchers(clock),
 		watchBookmarkEnabled: utilfeature.DefaultFeatureGate.Enabled(features.WatchBookmark),
 	}
-	watchCache.SetOnEvent(cacher.processEvent)
+
+	// Ensure that timer is stopped.
+	if !cacher.timer.Stop() {
```

Comment:
Should we need keep these code ? Have any reason to keep?

Reply:
Yes - we need this code, because we create above time.NewTimer(0).
So in order to not have any element in the .C channel, we need to consume it here.